### PR TITLE
Add more configuration options to cluster's node pools

### PR DESCRIPTION
**Pull request by oleksiimorozenko**
This PR adds configurable node labels to both system and user node pools
Also, it enables custom Kubernetes versions for user node pools if available node types versions differ from cluster version
Lastly, it adds an option for creating spot node pools by configuring `priority = "Spot"` option (and `"kubernetes.azure.com/scalesetpriority" = "spot"` node label)

### CODE REVIEW

**Comment by BzSpi** [comment]
Hello,

Thank you for your contribution, this is very well appreciated.

Regarding the content of the PR, either you should define the default value for new attributes in `local.default_agent_profile` or use `lookup` to access the value.

**Comment by oleksiimorozenko** [comment]
Default locals' values would be better I guess

**Comment by oleksiimorozenko** [comment]
This is the final version so far, tested, works correctly

**Review by Shr3ps — APPROVED** [review]

File: variables.tf [hunk 1]
```diff
@@ -265,7 +264,7 @@ variable "docker_bridge_cidr" {
 variable "nodes_pools" {
   description = "A list of nodes pools to create, each item supports same properties as `local.default_agent_profile`"
   type        = list(any)
```
Comment (jmapro) [inline variables.tf:266]: you may use a list of objects with the same content as for `default_node_pool `

**Review by jmapro — CHANGES_REQUESTED** [review]

**Comment by oleksiimorozenko** [comment]
Any news on this?

**Review by jmapro — APPROVED** [review]
LGTM

**Pull request merged**